### ISSUE TITLE
Swift3 - Why no more explicit overrideModule

### ISSUE THREAD

**Issue by duribreux** [square/Cleanse#38]
I used to have one `MainModule` with all my bindings inside and it was quite simple to override with Mock in my tests cases. Like:

```
propertyInjector = try! Component().withOverrides(overrideModule: FakeModule()).build()
```

And within my FakeModule
```
binder
    .bind(Service.self)
    .to(value: MockService())
```

Now I just can't do that anymore because of `Already bound` exception occurring when I install my `MainModule` and then try to override like I used to.

Is the only solution to split `MainModule` to more specifics `<Specific>Module` and only override the ones I need like you do in [AppComponent.swift](https://github.com/square/Cleanse/blob/swift-3/Examples/CleanseGithubBrowser/CleanseGithubBrowser/AppComponent.swift) or is there another way to override ?

**Comment by mikelikespie** [comment]
So we removed override modules for similar reasons dagger2 removed them.

However, when we removed them we didn't replace it with something to exchange components.

I have a WIP PR right here that  should improve this. https://github.com/square/Cleanse/pull/39/files#diff-1903bf857f2a6dd1ca32b4da88155b5e

Basically it allows one to have a "contract" with modules. 

I haven't updated this to work with the swift3 branch, but will do so after I get some feedback on the APIs.
